packages/truffle-core/lib/services/analytics/google.js

```diff
@@ -0,0 +1,168 @@
+    default: false
+  }
+];
+const analyticsEnable = [
+  {
+    type: "confirm",
+    name: "analyticsEnable",
+    message: "Analytics are currently disabled. Would you like to enable them?",
+    default: false
+  }
+];
+
+const googleAnalytics = {
+  /**
+   * set user-level unique id
+   */
+  setUserId: function() {
+    if (!userConfig.get("uniqueId")) {
+      let userId = nanoid();
+      userConfig.set({ uniqueId: userId });
+    }
+  },
+  /**
+   * set user-level options for analytics
+   * @param {bool} analyticsBool
+   * @param {Object} userConfig
+   */
+  setAnalytics: function(analyticsBool) {
+    if (analyticsBool === true) {
+      this.setUserId();
+      userConfig.set({
+        enableAnalytics: true,
+        analyticsSet: true,
+        analyticsMessageDateTime: Date.now()
+      });
+    } else {
+      userConfig.set({
+        enableAnalytics: false,
+        analyticsSet: true,
+        analyticsMessageDateTime: Date.now()
+      });
+    }
+    return true;
+  },
+  /**
+   * prompt user to determine values for user-level analytics config options
+   * @param {Object} userConfig
+   */
+  setUserConfigViaPrompt: async function() {
+    if (!userConfig.get("analyticsSet") && process.stdin.isTTY === true) {
+      let answer = await inquirer.prompt(analyticsInquiry);
+      if (answer.analyticsInquiry === analyticsInquiry[0].choices[0]) {
+        this.setAnalytics(true);
+      } else {
+        this.setAnalytics(false);
+      }
+    } else if (
+      userConfig.get("analyticsSet") &&
+      userConfig.get("enableAnalytics") &&
+      process.stdin.isTTY === true
+    ) {
+      let answer = await inquirer.prompt(analyticsDisable);
+      if (answer.analyticsDisable) {
+        this.setAnalytics(false);
+      } else {
+        this.setAnalytics(true);
+      }
+    } else if (
+      userConfig.get("analyticsSet") &&
+      !userConfig.get("enableAnalytics") &&
+      process.stdin.isTTY === true
+    ) {
+      let answer = await inquirer.prompt(analyticsEnable);
+      if (answer.analyticsEnable) {
+        this.setAnalytics(true);
+      } else {
+        this.setAnalytics(false);
+      }
+    }
+    return true;
+  },
+  /**
+   * check user-level config to see if user has enabled analytics
+   * @returns {bool}
+   */
+  checkIfAnalyticsEnabled: function() {
+    if (userConfig.get("enableAnalytics")) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+
+  /**
+   * set data that will be the same in future calls
+   * @returns {Object} visitor
+   */
+  setPersistentAnalyticsData: function() {
+    let visitor = false;
+    if (this.checkIfAnalyticsEnabled() === true) {
+      let userId = userConfig.get("uniqueId");
+      let visitor = ua(truffleAnalyticsId, { uid: userId });
+    }
+    return visitor;
+  },
+
+  /**
+   * send event to Google Analytics
+   * @param {Object}
+   */
+  sendAnalyticsEvent: function(eventObject) {
```

Comment:
Can we make this function `async` and return true or false based on whether the event was actually sent to GA?